jira/client.py

```diff
@@ -3705,19 +3705,23 @@ def enhanced_search_issues(
         Returns:
             Union[Dict, ResultList]: JSON Dict if ``json_result=True``, otherwise a `ResultList`.
         """
-        if isinstance(fields, str):
-            fields = fields.split(",")
-        elif fields is None:
+        if fields is None:
             fields = ["*all"]
+        elif fields and isinstance(fields, str):
+            fields = fields.split(",")
+        else:
+            # this is required only for mypy validation
+            fields = []
```

Comment:
(non-blocking) in general for these simple comments that don't put us above the line limit, I would push them onto one line, as these comments can get missed and become out of date when not associated with a line of code.

```suggestion
            fields = []  # help mypy
```